logging/jul/src/main/java/io/helidon/logging/jul/HelidonJsonFormatter.java

```diff
@@ -0,0 +1,280 @@
+/*
+ * Copyright (c) 2024 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.logging.jul;
+
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.logging.Formatter;
+import java.util.logging.LogManager;
+import java.util.logging.LogRecord;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import io.helidon.logging.common.HelidonMdc;
+import io.helidon.metadata.hson.Hson;
+
+import static io.helidon.logging.jul.HelidonFormatter.JUL_FORMAT_PROP_KEY;
+import static io.helidon.logging.jul.HelidonFormatter.THREAD_TOKEN;
+
+/**
+ * A {@link java.util.logging.Formatter} that stores each log record as a single-line JSON value.
+ * It also replaces all occurrences of MDC tags like {@code %X{value}} with specific values,
+ * and supports replacement of {@code "!thread!"} with the current thread.
+ * <p>
+ * The configuration should be done through property {@value #JSON_FORMAT_PROP_KEY}, that provides comma separated list of name to
+ * fields to log (references are the same as you would use in a {@link java.util.logging.SimpleFormatter}).
+ * <p>
+ * The configuration falls back to property {@value io.helidon.logging.jul.HelidonFormatter#JUL_FORMAT_PROP_KEY},
+ * and analyzes it to provide a "guessed" JSON structure.
+ * <p>
+ * Example (and also the default format):<br>
+ * {@value #DEFAULT_FORMAT}
+ */
+public class HelidonJsonFormatter extends Formatter {
+    static final String DEFAULT_FORMAT = "ts:%1$tQ,date:%1$tY.%1$tm.%1$td,time:%1$tH:%1$tM:%1$tS.%1$tL,level:%4$s,message:%5$s,"
+            + "exception:%6$s,thread:!thread!,logger:%3$s";
```

Comment:
For default format, I think date and time is more than enough and ts that represents milliseconds since the beginning of the epoch will not be necessary in general.

Reply:
Makes sense.